### PR TITLE
Enable LOGI and error logs in release builds.

### LINKED ISSUE #679: Oboe logs too much
Oboe Version: 1.2

**Steps To Reproduce**

Launch OboeTester
Click TEST OUTPUT
Click OPEN
Click START

**Expected behavior**

Few logs.

**Actual behavior**

Too many logs.

10-23 07:05:47.198 10214  4102  4102 I TestOboe: create OboeAudioOutputStream ---------
10-23 07:05:47.199 10214  4102  4102 I TestOboe: setToneType(1)
10-23 07:05:47.212 10214  4102  4102 D OboeTester: setActivityType(0)
10-23 07:05:49.501 10214  4102  4102 D OboeTester: OboeAudioStream_openNative: sampleRate = 0, framesPerBurst = 96
10-23 07:05:49.501 10214  4102  4102 D OboeAudio: ActivityContext::open() set callback to use oboeCallbackProxy, size = 0
10-23 07:05:49.501 10214  4102  4102 D OboeAudio: openStream() OUTPUT -------- OboeVersion1.2.2 --------
10-23 07:05:49.501 10214  4102  4102 D OboeAudio: AAudioLoader():  dlopen(libaaudio.so) returned 0x72a582035706c9f7
10-23 07:05:49.502 10214  4102  4102 D OboeAudio: AudioStreamAAudio() call isSupported()
10-23 07:05:49.528 10214  4102  4102 D OboeAudio: AudioStreamAAudio.open() app    format = 2
10-23 07:05:49.528 10214  4102  4102 D OboeAudio: AudioStreamAAudio.open() sample rate   = 48000
10-23 07:05:49.528 10214  4102  4102 D OboeAudio: AudioStreamAAudio.open() capacity      = 3072
10-23 07:05:49.528 10214  4102  4102 D OboeAudio: AudioStreamAAudio.open: AAudioStream_Open() returned AAUDIO_OK, mAAudioStream = 0x7bba4c6700
10-23 07:05:49.528 10214  4102  4102 D OboeAudio: ActivityContext::open() builder.openStream() returned 0
10-23 07:05:49.528 10214  4102  4102 D OboeTester: OboeAudioStream_getNativeApi got 2
10-23 07:05:50.819 10214  4102  4102 I TestOboe: startAudio() called =======================================
10-23 07:05:50.819 10214  4102  4102 D OboeAudio: ActivityContext: start() called
10-23 07:05:51.129 10214  4102  4702 D OboeAudio: AudioStream::fireDataCallback() scheduler = SCHED_FIFO
10-23 07:05:51.129 10214  4102  4102 I TestOboe: setToneType(1)
10-23 07:05:57.373 10214  4102  4102 D OboeAudio: ActivityContext::stopAllStreams() called
10-23 07:05:57.373 10214  4102  4102 D OboeAudio: ActivityContext::stopAllStreams() handle = 0, stream 0x7acb3fdee0
10-23 07:05:57.428 10214  4102  4102 I OboeAudio: scheduler = 0x40000001, SCHED_FIFO = 0x00000001
10-23 07:05:58.485 10214  4102  4102 D OboeAudio: ActivityContext::close() delete stream 0

### LINKED ISSUE #866: Enable critical Oboe logging for release builds
Oboe logging is only enabled for debug builds. So we are not seeing version numbers or error logs for release apps. That makes it hard to know if Oboe is getting called. 

From  "oboe/CMakeLists.txt":

    # Enable logging for debug builds
    target_compile_definitions(oboe PUBLIC $<$<CONFIG:DEBUG>:OBOE_ENABLE_LOGGING=1>)

We should only disable LOGD and LOGV.
We should always log LOGI, LOGE, LOGW and LOGF.

And we should remove any unnecessary logs.

### PR DESCRIPTION
**Pull request by philburk**
Fixes #866 
Fixes #679

### CODE REVIEW

File: src/common/AudioStream.cpp [hunk 1]
```diff
@@ -39,22 +39,24 @@ Result AudioStream::close() {
...
-    // TODO remove
+// Call this from fireDataCallback to monitor CPU scheduler.
+void AudioStream::checkScheduler() {
     int scheduler = sched_getscheduler(0) & ~SCHED_RESET_ON_FORK; // for current thread
     if (scheduler != mPreviousScheduler) {
         LOGD("AudioStream::%s() scheduler = %s", __func__,
              ((scheduler == SCHED_FIFO) ? "SCHED_FIFO" :
-             ((scheduler == SCHED_OTHER) ? "SCHED_OTHER" :
-             ((scheduler == SCHED_RR) ? "SCHED_RR" : "UNKNOWN")))
+              ((scheduler == SCHED_OTHER) ? "SCHED_OTHER" :
```
Comment (android-usb) [inline src/common/AudioStream.cpp:48]: nit: is this the indentation you wanted?

File: src/common/AudioStream.cpp [hunk 2]
```diff
@@ -39,22 +39,24 @@ Result AudioStream::close() {
...
+               ((scheduler == SCHED_RR) ? "SCHED_RR" : "UNKNOWN")))
         );
         mPreviousScheduler = scheduler;
     }
+}
+
+DataCallbackResult AudioStream::fireDataCallback(void *audioData, int32_t numFrames) {
+    if (!isDataCallbackEnabled()) {
+        LOGW("AudioStream::%s() called with data callback disabled!", __func__);
```
Comment (android-usb) [inline src/common/AudioStream.cpp:57]: Maybe line length?

File: src/common/AudioStream.cpp (hunk 1 above)
Comment (philburk) [reply src/common/AudioStream.cpp:48]: Nope. Auto-indenter got creative. Thanks.

File: src/common/AudioStream.cpp (hunk 2 above)
Comment (philburk) [reply src/common/AudioStream.cpp:57]: Line 55 has 87 characters. Limit according to Android style guide is 100. Seems OK.

**Comment by android-usb** [comment]
Ack. Merge that sucka!

**Comment by philburk** [comment]
Is there any possibility that adding logging to a release build will cause a linker problem?

**Comment by dturner** [comment]
> Is there any possibility that adding logging to a release build will cause a linker problem?

I don't believe so. Additionally, linker problems should be caught by CI. [Build is passing for this commit](https://travis-ci.org/github/google/oboe/builds/691916739)

**Pull request merged**